x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/multi_metric_job_creator.ts

```diff
@@ -90,7 +100,13 @@ export class MultiMetricJobCreator extends JobCreator {
     this._overrideConfigs(job, datafeed);
     this.createdBy = CREATED_BY_LABEL.MULTI_METRIC;
     this._sparseData = isSparseDataJob(job, datafeed);
-    const detectors = getRichDetectors(job, datafeed, this.additionalFields, false);
+    const detectors = getRichDetectors(
```

Comment:
I think `getRichDetectors` (and `getFieldByIdFactory`) could be moved to be a member of the base class. It then wouldn't need `newJobCapsService` to be passed to it.

I can do this in a follow up, if you don't want to complicate this PR any further.

Reply:
Yes, let's do this in follow ups please 😅 .